test/suites/basic.sh

```diff
@@ -542,4 +541,21 @@ test_basic_usage() {
   # Test renaming/deletion of the default profile
   ! lxc profile rename default foobar || false
   ! lxc profile delete default || false
+
+  # Test profile not allowed to have instance specific config keys of different types
+  lxc profile create instanceTypeTest
+  lxc profile set instanceTypeTest security.privileged true
+  ! lxc profile set instanceTypeTest security.secureboot true || false
+  lxc profile delete instanceTypeTest
+
+  # Test instance type specific config keys
+  lxc init testimage instance-type-specific-config-test-container
+  lxc init testimage instance-type-specific-config-test-vm
```

Comment:
Sadly we cannot create VMs are part of the test suite.

Reply:
ok, i removed the faulty vm test.

Reply:
Let me know when the other issues (like go vet) are resolved and I'll re-run the tests. Thanks